### PR TITLE
add descriptions and other annotations to extensions

### PR DESCRIPTION
**Pull request by turbomam**

### CODE REVIEW

**Review by sujaypatil96 — APPROVED** [review]

File: src/mixs/schema/mixs.yaml [hunk 1]
```diff
@@ -16961,8 +16997,18 @@ classes:
         examples:
         - value: 1.6 kilometers per hour
     class_uri: MIXS:0016020
+    aliases:
+      - MIxS-Food (farm environment)
+    annotations:
+      use_cases: Microbiome of farm and field crops as well as environmental samples including irrigation, soil amendments, and farm equipment.
   FoodFoodProductionFacility:
```
Comment (mslarae13) [inline src/mixs/schema/mixs.yaml:17004]: I think we can make a separate issue for this.. but is it really foodfood? Why is food twice?

**Review by mslarae13 — APPROVED** [review]

File: src/mixs/schema/mixs.yaml (hunk 1 above)
Comment (turbomam) [reply src/mixs/schema/mixs.yaml:17004]: Good question. That is not a typo or bug. In this case, it is helpful to look at the submitter-provided `title` of this class: 'food-food production facility'. Pre-LinkML, an attempt was made to **approximate** hierarchy in the `Extension`s, as if there is a category of "food" `Extension`s, of which 'food production facility' is a child `Extension`. 

Following [standard naming for LinkML classes](https://linkml.io/linkml/schemas/linter.html#standard-naming), the `title` `food-food production facility` has been transformed into a term `name` of 'FoodFoodProductionFacility'.

I don't have any objection t changing that, but it should be done systematically.

Comment (only1chunts) [reply src/mixs/schema/mixs.yaml:17004]: @turbomam is correct, it was an attempt at hierarachy, the FDA were involved in designing a set of 4 extensions, so they grouped them all under a "Food" heading. 

- MIxS-food-animal and animal feed
- MIxS-food-farm environment
- MIxS-food-food production facility
- MIxS-food-human foods

I'm not against renaming them, but there maybe parties that might be opposed to it, so I have opened a discussion ticket for it #779

File: src/mixs/schema/mixs.yaml [hunk 2]
```diff
@@ -16354,8 +16358,13 @@ classes:
...
         recommended: true
     class_uri: MIXS:0016018
+    aliases:
+      - MIxS-Ag (agriculture)
+    annotations:
+      use_cases: Agricultural Microbiomes Research Coordination Network, model cropping and plant systems focused on agricultural plant and soil microbe; microbiome studies in agricultural sites; long-term ecological research in croplands; eDNA in manure samples; describing agricultural microbiome studies
   Air:
-    description: air extension
+    description: >
+      A collection of terms appropriate when collecting and sequencing obtained from a gaseous environment.
```
Comment (only1chunts) [inline src/mixs/schema/mixs.yaml:16367]: suggested change: A collection of terms appropriate when collecting and sequencing **samples** obtained from a gaseous environment.

File: src/mixs/schema/mixs.yaml [hunk 3]
```diff
@@ -17473,8 +17553,14 @@ classes:
...
     class_uri: MIXS:0016003
+    aliases:
+      - MIxS-human-associated
+    annotations:
+      use_cases: blood samples or biopsy samples.
   HumanGut:
-    description: human-gut extension
+    description: >
+      A collection of terms appropriate when collecting samples and sequencing samples 
+      obtained from a person to examine their oral-associated microbiome.
```
Comment (only1chunts) [inline src/mixs/schema/mixs.yaml:17563]: typo - "oral-associated" should be "gut-associated"

File: assets/mixs-schemasheets-concise-report.txt
Comment (only1chunts) [inline assets/mixs-schemasheets-concise-report.txt:1]: I'm not sure what this file is, or how it relates to the updating of extension descriptions, but I trust its meant to be here?

File: assets/mixs-schemasheets-concise-log.txt
Comment (only1chunts) [inline assets/mixs-schemasheets-concise-log.txt:1]: unable to view this as github says the diff is too large!

File: assets/mixs-patterns-materialized.yaml
Comment (only1chunts) [inline assets/mixs-patterns-materialized.yaml:1]: unable to voiew this as github says the diff is too large.

File: assets/isolate_slots.py
Comment (only1chunts) [inline assets/isolate_slots.py:1]: I'm not sure what this file is, or how it relates to the updating of extension descriptions, but I trust its meant to be here?

File: assets/class_summary_results.tsv
Comment (only1chunts) [inline assets/class_summary_results.tsv:1]: I'm not sure what this file is, or how it relates to the updating of extension descriptions, but I trust its meant to be here?

File: assets/mixs-schemasheets-concise-report.txt
Comment (turbomam) [reply assets/mixs-schemasheets-concise-report.txt:1]: This is one of may outputs from running the `linkml2schemasheets-template` command in a way that's similar to but not identical with the [assets/mixs_derived_class_term_schemasheet.tsv](https://github.com/GenomicsStandardsConsortium/mixs/blob/573e56a35d11a2062d35f0f7fc58f296f3205298/project.Makefile#L20-L26) target in the project.Makefile

That process generates a tabular overview of the schema. Similar processes int eh past required the user to provide a template indicating which attributes of which element types should be reported, but a template is not required for `linkml2schemasheets-template`.

File: assets/class_summary_results.tsv
Comment (turbomam) [reply assets/class_summary_results.tsv:1]: I have updated `project.Makefile` to [demonstrate how this file is built.](https://github.com/GenomicsStandardsConsortium/mixs/blob/47079d92af73c282428b9a11f842ca47e3f175f7/project.Makefile#L40-L43)

The result is a file that we discussed a few weeks ago: a tabular representation of textual annotations on the MIxS classes (Checklists, Extensions and Combinations)

Comment (turbomam) [reply assets/class_summary_results.tsv:1]: https://github.com/GenomicsStandardsConsortium/mixs/blob/646-classes-checklists-and-extensions-require-improved-descriptions/assets/class_summary_results.tsv

File: assets/mixs-schemasheets-concise.tsv
[the remark this answers is not in the document]
Comment (turbomam) [reply assets/mixs-schemasheets-concise.tsv:1]: This file, and the following status logs, are is generated by the the [assets/mixs-schemasheets-concise.tsv target in project.Makefile](https://github.com/GenomicsStandardsConsortium/mixs/blob/c7cd23bb476f5d40e8f76bb883e446aaa9650926/project.Makefile#L45-L51)

File: assets/mixs-schemasheets-concise-log.txt
Comment (turbomam) [reply assets/mixs-schemasheets-concise-log.txt:1]: See notes on [assets/mixs-schemasheets-concise.tsv](https://github.com/GenomicsStandardsConsortium/mixs/pull/769/files/fa4d1829faf027e53bec5673d72cc87aa45b8071#diff-8608cac67ebf1c207b3ea1c0f5f6bfcbef20e249a60cbcbd6bddf54c9e3efe9f)

File: assets/isolate_slots.py
Comment (turbomam) [reply assets/isolate_slots.py:1]: LinkML provides two mechanisms for applying a regular expression constraint to the values of a term/slot.
- the `pattern` meta-slot, in which you have to provide the exact regular expression that you consider valid for that slot
- the `structured_pattern` meta-slot, which allows you to compose the regular expression from previously-defined building blocks in the schema's `settings` section.

File: assets/mixs-patterns-materialized.yaml
Comment (turbomam) [reply assets/mixs-patterns-materialized.yaml:1]: I had added a [project.Makefile target for regenerating it](https://github.com/GenomicsStandardsConsortium/mixs/blob/bfa4771924b1f30c125327a5c483c876d4147709/project.Makefile#L53-L55) that illustrates how this file is generated.

LinkML provides two mechanisms for applying a regular expression constraint to the values of a term/slot.
- the `pattern` meta-slot, in which you have to provide the exact regular expression that you consider valid for that slot
- the `structured_pattern` meta-slot, which allows you to compose the regular expression from previously-defined building blocks in the schemas's `settings` section.

To see how the `structured_pattern`s are converted to regular expressions, you need to run the schema through some kind of materializing generator.

File: assets/isolate_slots.py
Comment (turbomam) [reply assets/isolate_slots.py:1]: This script extracts the global definitions of MIxS terms/slots from XXX. "Global" mean that the output does not indicate the cases in which one Extension class may. It also does a little more convenience filtering:
- removes the schemasheet header rows that start with a `>` character
- removes rows about classes
- removes rows about slots used by the purely infrastructural class `MixsCompliantData`, which nis used for validating tables of data that is believed to follow MIxS guidelines
- removes any columns if they don't contain any data values 

I have commented out some lines that can insert the filtered table back into a relational database. This is especially useful for NMDC but I am glad to teach other people how to use it.

File: assets/mixs-schemasheets-concise-global-slots.tsv
[the remark this answers is not in the document]
Comment (turbomam) [reply assets/mixs-schemasheets-concise-global-slots.tsv:1]: see comments about [assets/isolate_slots.py](https://github.com/GenomicsStandardsConsortium/mixs/pull/769/files/fa4d1829faf027e53bec5673d72cc87aa45b8071#diff-5abddc18cd67959f5f90b03997da97dccfe1fc6f718f3c145b0ce80e5684bdb5)

**Comment by turbomam** [comment]
Good questiosn @only1chunts

**Comment by turbomam** [comment]
Discussed in [CIG meeting 2024-03-26](https://docs.google.com/document/d/19CWWf1oqMlyH7prteVC5k4eYF_JzJbNqNcvGUyX_U50/edit)

Thanks for the feedback. I will stick to smaller PRs next time!

Followup actions:
- integrate contributor and reference knowledge from
- work on separate release artifact directory (just a renaming of `project/`?) What files are required, in terms of structure and content?
- come to an agreement on structured formats for sharing attributes of entities (like Extensions). Something other than [Google (word processing) Docs](https://docs.google.com/document/d/1km8P37kM1R9_AWlDnTeWq2HaQ3EA6X0djzBWn9bJL2A/edit#headi)

**Pull request merged**